firebase-firestore/src/test/java/com/google/firebase/firestore/model/TargetIndexMatcherTest.java

```diff
@@ -802,20 +804,32 @@ public void testBuildTargetIndexWithInAndOrderBy() {
             .filter(filter("a", "not-in", Arrays.asList(1, 2, 3)))
             .orderBy(orderBy("a"))
             .orderBy(orderBy("b"));
-    validateBuildTargetIndexCreateFullMatchIndex(query);
+    validateBuildTargetIndexCreateFullMatchIndexForSingleInequality(query);
   }
 
   @Test
   public void testBuildTargetIndexWithInAndOrderBySameField() {
     Query query =
         query("collId").filter(filter("a", "in", Arrays.asList(1, 2, 3))).orderBy(orderBy("a"));
-    validateBuildTargetIndexCreateFullMatchIndex(query);
+    validateBuildTargetIndexCreateFullMatchIndexForSingleInequality(query);
   }
 
-  private void validateBuildTargetIndexCreateFullMatchIndex(Query query) {
+  @Test
+  public void testBuildTargetIndexReturnsNullForMultipleInequality() {
+    Query query = query("collId").filter(filter("a", ">=", 1)).filter(filter("b", "<=", 10));
+    Target target = query.toTarget();
+    TargetIndexMatcher targetIndexMatcher = new TargetIndexMatcher(target);
+    assertTrue(targetIndexMatcher.hasMultipleInequality());
+    FieldIndex expectedIndex = targetIndexMatcher.buildTargetIndex();
+    assertNull(expectedIndex);
+  }
+
+  private void validateBuildTargetIndexCreateFullMatchIndexForSingleInequality(Query query) {
```

Comment:
I think the rename is not an improvement for readability. Can we revert the name back?